lib/App/DuckPAN/Perl.pm

```diff
@@ -109,54 +111,73 @@ sub duckpan_install {
 		$sp =~ s/\:\:/_/g;
 
 		# special case: check for a pinned verison number
-		my $pin_version            = $ENV{$sp};
-		my $localver               = $self->get_local_version($package);
-		my $duckpan_module_version = version->parse($module->version);
-		my $duckpan_module_url     = $self->app->duckpan . 'authors/id/' . $module->distribution->pathname;
+		my $pinned_version      = $ENV{$sp};
+		my $installed_version   = $self->get_local_version($package);
+		my $latest_version      = version->parse($module->version);
+		my $duckpan_module_url  = $self->app->duckpan . 'authors/id/' . $module->distribution->pathname;
 
-		$localver ||= 1e-6 if ($pin_version); # a silly, but true, value if missing and we need to compare with pinned.
```

Comment:
>You're talking about 9.999, right? 

Nope. If I dzil install the repo because I'm developing in it then my installed version of the repo would be 1 higher than latest. So IIRC `duckpan upgrade` won't have any effect as the installed version is higher than the currently released/pinned version.